lib/galaxy/util/xml_util.py

```diff
@@ -115,28 +124,23 @@ def indent( elem, level=0 ):
             elem.tail = i
 
 
-def parse_xml( file_name ):
+def parse_xml( file_name, preserve_comments=False ):
     """Returns a parsed xml tree with comments intact."""
-    error_message = ''
-    fobj = open( file_name, 'r' )
-    if using_python_27:
-        try:
-            tree = XmlET.parse( fobj, parser=XmlET.XMLParser( target=Py27CommentedTreeBuilder() ) )
-        except Exception, e:
-            fobj.close()
-            error_message = "Exception attempting to parse %s: %s" % ( str( file_name ), str( e ) )
-            log.exception( error_message )
-            return None, error_message
-    else:
+    with open( file_name, 'r' ) as fh:
+        if sys.version_info[ :2 ] >= ( 2, 7 ):
+            xml_parser = XmlET.XMLParser( target=Py27CommentedTreeBuilder() )
+        else:
+            xml_parser = Py26CommentedTreeBuilder()
         try:
-            tree = XmlET.parse( fobj, parser=Py26CommentedTreeBuilder() )
+            if preserve_comments:
+                tree = XmlET.parse( fh, parser=xml_parser )
+            else:
+                tree = XmlET.parse( fh )
         except Exception, e:
-            fobj.close()
             error_message = "Exception attempting to parse %s: %s" % ( str( file_name ), str( e ) )
             log.exception( error_message )
             return None, error_message
-    fobj.close()
-    return tree, error_message
+    return tree, None
```

Comment:
I was wondering: would it be cleaner/more Pythonic to raise an Exception in case of error, instead of always returning a (potentially empty) error message?